django/views/csrf.py

```diff
@@ -131,4 +130,8 @@ def csrf_failure(request, reason=""):
         'docs_version': get_docs_version(),
         'more': _("More information is available with DEBUG=True."),
     })
+    try:
+        t = loader.get_template(template_name)
+    except TemplateDoesNotExist:
+        t = Engine().from_string(CSRF_FAILURE_TEMPLATE)
```

Comment:
I think we shouldn't silence the exception in all cases here, only when `template_name != 'csrf_failure.html'`.

Changing the default value of `template_name` to `None` and actually raising the exception `if template_name is not None` should do.